grammar.js

```diff
@@ -62,6 +62,10 @@ module.exports = grammar({
     [$._unannotated_type, $.scoped_type_identifier],
     [$._unannotated_type, $.generic_type],
     [$.generic_type, $.primary_expression],
+    // Switch constructs can be both expressions and statements
+    [$.expression, $.statement],
+    // Only conflicts in switch expressions
+    [$.lambda_expression, $.primary_expression],
```

Comment:
Do you think it's allowed to have a lambda expression within a switch rule? If not, maybe we could use precedence to remove this conflict as well.

Reply:
I don't think so? Looking at https://docs.oracle.com/javase/specs/jls/se16/html/jls-19.html, we do see that LambdaExpression can be used in a CastExpression, which can end up as a ConditionalExpression (this is what Java limits its case label to). However, from a type matching perspective, I don't believe a LambdaExpression can ever appear in a valid case label, since we can't case on lambdas, and it seems like Java doesn't support lambdas for function application `((x -> x)(2))`. I'm not super familiar with lambdas, though, so I might be missing something.